scalableminds/webknossos
# Unsigned Access to Pubilc S3 Data

I had to make two changes in the s3fs library. To do that, I copied it into webKnossos (and added the original MIT license in the package). I’ll mark the relevant changes I made to the library below to make the review easier.

### URL of deployed dev instance (used for testing):
- https://s3unsigned.webknossos.xyz

### Steps to test:
- In add remote zarr dataset view, paste s3 uri with no credentials, e.g. `s3://power-analysis-ready-datastore.s3.amazonaws.com/power_901_monthly_meteorology_utc.zarr/EVLAND`, click Add Layer, should yield a datasource
- Also try abbreviated s3 syntax, e.g. `s3://power-analysis-ready-datastore/power_901_monthly_meteorology_utc.zarr/EVLAND` (no `s3.amazonaws.com`), should also work
- Also make sure that s3 access _with_ credentials still works

### Issues:
- fixes #6401 

Note that this also includes the changes of #6422 because I needed them for testing. I’ll take care of merge conflicts if they come up.

------
